### PR TITLE
feat: pass additional arguments to scripts

### LINKED ISSUE #7: Feature request: Pass additional flags to spawned commands
Example setup:

```json
{
  "scripts": {
    "test": "is-ci \"test:ci\" \"test:local\"",
    "test:ci": "jest --config jest.ci.config.js",
    "test:local": "jest --config jest.local.config.js"
  }
}
```

I want to be able to pass additional flags to my `jest` wrappers locally and in CI:

```bash
yarn test --watch 
# jest --config jest.local.config.js --watch
```

```bash
export CI=1
yarn test --coverage --coverageReporters=text-summary
# jest --config jest.ci.config.js --coverage --coverageReporters=text-summary
```

### PR DESCRIPTION
**Pull request by KubaJastrz**
resolves #7

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,5 +34,10 @@
     "ci:test": "is-ci ci:pass ci:fail",
     "ci:fail": "exit 1",
     "ci:pass": "exit 0"
+  },
+  "xo": {
+    "rules": {
+      "camelcase": "off"
```
Comment (KubaJastrz) [inline package.json:40]: needed for `process.env.npm_execpath` to be reassigned in tests

**Comment by KubaJastrz** [comment]
Good idea about the 3rd argument. I'll take a look at the Windows tests too. Thanks for the heads up!

**Comment by KubaJastrz** [comment]
I followed your suggestion about the 3rd argument. It's looking much better now.

As for the Windows tests - it seems like there's a difference between how `childProcess.spawn` is [treated on Windows](https://github.com/moxystudio/node-cross-spawn/blob/7501971226a7c8b2cf0c1acd1703547e3798491b/lib/parse.js#L57): 

```js
// Windows
spawn('yarn', ['run', 'test']) // => "\"yarn ^\"run^\" ^\"test^\"\""

// Unix
spawn('yarn', ['run', 'test']) // => ['yarn', 'run', 'test']
```

I think we just need to spy on higher abstraction (`cross-spawn` instead of `child_process`).

**Comment by YellowKirby** [comment]
> I think we just need to spy on higher abstraction (cross-spawn instead of child_process).

Sounds good to me. When those tests were first written I wanted it to cover more of the overall use and verify that we were doing the right thing all the way down to the process getting spawned. But #6 added a pretty clever way of having better functional tests for this package, so I don't think it's as important that we test the spawn part in what should be unit tests anymore.

So maybe we could break the functions up a bit more:

1. Smaller pure function that just processes the args (basically what `run` is now but without calling out to `cross-spawn`). `test.js` primarily deals with this function.
2. More side-effecty function that deals with the `npm_execpath` environment variable, calling `cross-spawn`, etc that is covered by the tests added in #6. Maybe even expand on these and add some arguments?

What do you think?

Either way I think a lot of this could be a different PR, so maybe just do what's easy/simple to get the Windows stuff passing here?

**Comment by KubaJastrz** [comment]
Yes, I like the suggestions. I'm gonna work on these in the upcoming days.

I'm having some issues with spying on the `import spawn from 'cross-spawn'` as the `sinon` requires an object and a property to spy on. Is there an equivalent of `jest.mock('module')` in `sinon`? I'm not familiar with `sinon` that much.

**Comment by YellowKirby** [comment]
As far as I know there really isn't a great way to handle stubbing out default exports. The stub needs to be in place before the first call to `require('cross-spawn')`, which is surprisingly difficult to set up given that there shouldn't be any code above an `import` statement in a module.

cross-spawn does export the same `spawn` function as a named property instead of just the default: https://github.com/moxystudio/node-cross-spawn/blob/879eb84a99fe7845557aa484d591e3cd08254ed3/index.js#L35, so the following should work:

```js
// cli.js
const crossSpawn = require('cross-spawn');

function run() {
  return crossSpawn.spawn(...)
}
```

```js
// test.js
import crossSpawn from 'cross-spawn';

test.before('stub spawn', () => {
	sinon.stub(crossSpawn, 'spawn').callsFake((name, args) => args.slice(-2));
});
```

But it also might be simpler to just structure the functions that are being tested in a way that they don't rely on cross-spawn at all.

File: package.json [hunk 2]
```diff
@@ -24,7 +24,7 @@
   },
   "devDependencies": {
     "ava": "^0.19.1",
-    "sinon": "^2.2.0",
+    "sinon": "^9.0.2",
```
Comment (KubaJastrz) [inline package.json:27]: needed for `sinon.fake()`

**Comment by KubaJastrz** [comment]
> But it also might be simpler to just structure the functions that are being tested in a way that they don't rely on cross-spawn at all.

Yep, this was way simpler to accomplish. Tests should be passing now for all operating systems (tested on my Windows machine).

File: package.json (hunk 2 above)
Comment (KubaJastrz) [reply package.json:27]: Actually, I see that sinon v9 dropped support for node v8. Should we also do the same, or downgrade to sinon v8?

Comment (KubaJastrz) [reply package.json]: downgraded for the time being

**Comment by YellowKirby** [comment]
This was published in version [`2.1.0`](https://github.com/YellowKirby/is-ci-cli/releases/tag/v2.1.0). Thanks @KubaJastrz!

**Comment by KubaJastrz** [comment]
Thanks for helpful responses and great package!

**Pull request merged**